Client/lib/gMap.js

```diff
@@ -0,0 +1,231 @@
+
+    //----------------------------------
+    //events
+    //----------------------------------
+    google.maps.event.addListener(map, 'click', function(event) {
+      addMarker(event.latLng, 'Count ' + markerCords.length);
+    });
+
+    //----------------------------------
+    //adding custom GUI elements
+    //----------------------------------
+
+
+    var tagForZoomOut = document.createElement('div');
+    var controlForZoomOut = new CenterControl(tagForZoomOut, '-', function(){
+      map.setZoom(map.getZoom()-1);
+    });
+
+    var tagForZoomIn = document.createElement('div');
+    var controlForZoomIn = new CenterControl(tagForZoomIn, '+', function(){
+      map.setZoom(map.getZoom()+1);
+    });
+
+    var tagForWalk = document.createElement('div');
+    var controlForWalk = new CenterControl(tagForWalk, 'W', function(){
+      travelMode = 'WALKING';
+    },true,true);
+    
+    var tagForBicycle = document.createElement('div');
+    var controlForBicycle = new CenterControl(tagForBicycle, 'B', function(){
+      travelMode= 'BICYCLING';
+    },true);
+    
+    var tagForDrive = document.createElement('div');
+    var controlForDrive = new CenterControl(tagForDrive, 'D', function(){
+      travelMode= 'DRIVING';
+    },true);
+
+    var tagForTransit = document.createElement('div');
+    var controlForTransit = new CenterControl(tagForTransit, 'T', function(){
+      travelMode= 'TRANSIT';
+    },true);
+
+    //adding the events tags to the Map
+    
+    map.controls[google.maps.ControlPosition.TOP_LEFT].push(tagForZoomOut);
+    map.controls[google.maps.ControlPosition.TOP_LEFT].push(tagForZoomIn);
+    map.controls[google.maps.ControlPosition.BOTTOM_LEFT].push(tagForWalk);
+    map.controls[google.maps.ControlPosition.BOTTOM_LEFT].push(tagForBicycle);
+    map.controls[google.maps.ControlPosition.BOTTOM_LEFT].push(tagForDrive);
+    map.controls[google.maps.ControlPosition.BOTTOM_LEFT].push(tagForTransit);
+  }
+
+
+
+
+  //------------------------
+  //functions
+  //------------------------
+
+  function calcRoute() {
+
+    var directionsDisplay = new google.maps.DirectionsRenderer({suppressMarkers: true});
+    var directionsService = new google.maps.DirectionsService();
+  
+    directionsDisplay.setMap(map);
+
+    var start = markerCords[markerCords.length-2];
+    var end = markerCords[markerCords.length-1];
+    var request = {
+        origin:start,
+        destination:end,
+        travelMode: google.maps.TravelMode[travelMode],
+    };
+    directionsService.route(request, function(response, status) {
+      if (status == google.maps.DirectionsStatus.OK) {
+        directionsDisplay.setDirections(response);
+      }
+    });
+  }
+
+  function addMarker (latLng, title){
+    var id = markerCords.length;
+    markerCords.push(latLng);
+    var size = {
+      height: 40,
+      width: 40
+    }
+
+    var image = {
+      url: 'https://cdn2.iconfinder.com/data/icons/snipicons/500/map-marker-512.png',
```

Comment:
This image was never meant to be the final image once we find the final image it will be easy to change.